lamagraph-core/src/Core/Node.hs

```diff
@@ -0,0 +1,95 @@
+module Core.Node where
+
+import Clash.Prelude
+import Data.Maybe (isJust)
+
+type Address = Unsigned 16
+
+-- | Pointer on port of the `Node` with visited flag.
+data Port = Port
+  { _targetAddress :: Address
+  , _edgeIsVisited :: Bit
```

Comment:
In my opinion `Bit` should be used to represent signals that actually flow in circuit, like buttons, and `Bool` for logical value. 
Also, here custom type may be used to avoid boolean blindness.